components/ShowVotesByAddress.tsx

```diff
@@ -0,0 +1,98 @@
+import {
+  Button,
+  createStyles,
+  Group,
+  Paper,
+  Select,
+  Stack,
+  Table,
+  Title,
+} from "@mantine/core";
+import { BigNumber } from "ethers";
+import { useState } from "react";
+import { VotingContractState } from "../hooks/useVotingDapp";
+
+type ShowVotesByAddressProps = {
+  contractState: VotingContractState;
+  onSubmit: (voterAddress) => Promise<[BigNumber, BigNumber[]]>;
+};
+
+const useStyles = createStyles((theme) => ({
+  tableStyle: {
+    borderCollapse: "separate",
+    borderRadius: "12px",
+    border: "1px solid black",
+  },
+  selIp: {
+    flexGrow: 1,
+  },
+}));
+
+export default function ShowVotesByAddress({
+  contractState,
+  onSubmit,
+}: ShowVotesByAddressProps) {
+  const { classes } = useStyles();
+  const [voterAddress, setCandidate] = useState<string | null>(null);
```

Comment:
Unrelated to the web3 side of things - but React hooks best practice: Name your useState() hook return values as <var> and set<var>

eg. `const [voterAddress, setVoterAddress] = useState.....`

Reply:
The function voterDetails returns an array of the type [BigNumber, BigNumber[]] ... for some reason, i was getting an error where it looked at it as a Map of <hex, BigNumber>.. so i did that... maybe just passing the entire map is better right!